### DISCUSSION TITLE
Quarkus : Mock bean in dev mode

### DISCUSSION THREAD

**Discussion by antoinechamot** [quarkusio/quarkus#34269] [Community]
In my Quarkus application have a bean with a grpc client :
```java
@ApplicationScoped
@Slf4j
public class AuthorizationAdapter implements AuthorizationPort {

    private final AuthorizationGrpc.AuthorizationBlockingStub authorizationBlockingStub;

    public AuthorizationAdapter(
            @GrpcClient("authorization") AuthorizationGrpc.AuthorizationBlockingStub authorizationBlockingStub) {
        this.authorizationBlockingStub = authorizationBlockingStub;
    }

    @Override
    public boolean canActorReadReport(Actor actor, UUID reportId) {
        ... call authorizationBlockingStub here

    }
}
``` 
For test i managed to mock this bean with :
```java
@ApplicationScoped
@Mock
public class MockAuthorizationAdapter implements AuthorizationPort {

    @Override
    public boolean canActorReadReport(Actor actor, UUID reportId) {
        return true;
    }
}
``` 
But in dev mode i want to do the same (i don't have and want to start a grpc server). How can i do ? 
I am aware that i could annotate this bean with @IfBuildProfile("prod") and create another one mocked, but it would pollute my src/main/java and i don't want that. Is it possible to achieve this elegantly ?

**Comment by bulldog98** [comment]
I think your best bet would be to have an custom extension that starts the mock server and configure it to run for dev and testing. This also make that reusable if you have more microservices depending on that rpc server
